### PR TITLE
Add Pull Request template to the repository

### PR DESCRIPTION
**Pull request by scb01**
This commit adds a checklist for PR authors to follow. This initial rev of the PR checklist contains general documentation, testing guidelines, and points PR authors to our contribution guidelines.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+This checklist is used to make sure that common guidelines for a pull request are followed.
+
+### General Guidelines and Best Practices
+- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
+- [ ] Title of the pull request is clear and informative.
+- [ ] Description of the pull request includes a brief summary of the enhancement or bug fix.
+- [ ] There are a small number of commits, each with an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
+
+### Testing Guidelines
+- [ ] Description of the pull request includes a brief summary of tests added/modified and the local testing done.
```
Comment (damonbarry) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: I think this should be two separate items with their own checkboxes (1. brief summary of tests added/modified, 2, local testing done)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+### General Guidelines and Best Practices
+- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
+- [ ] Title of the pull request is clear and informative.
+- [ ] Description of the pull request includes a brief summary of the enhancement or bug fix.
+- [ ] There are a small number of commits, each with an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
+
+### Testing Guidelines
+- [ ] Description of the pull request includes a brief summary of tests added/modified and the local testing done.  
+- [ ] Pull request includes test coverage for the included changes.
```
Comment (damonbarry) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: I'd recommend this one go first in the Testing Guidelines section.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Azure IoTEdge PR checklist:
+
+This checklist is used to make sure that common guidelines for a pull request are followed.
+
+### General Guidelines and Best Practices
+- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
+- [ ] Title of the pull request is clear and informative.
+- [ ] Description of the pull request includes a brief summary of the enhancement or bug fix.
+- [ ] There are a small number of commits, each with an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
```
Comment (damonbarry) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: I don't fully understand this one. I think the statement about previously merged commits showing up in the PR probably goes without saying. And I think the powershell doc focuses on the wrong things, like # of commits (vs. # of concepts in the PR) and details like rebase vs. merge.

I do think PRs should be scoped to focus on a single thing (e.g., fixing a bug _and_ doing a codebase-wide reformatting of braces in one PR == bad), and split into multiple PRs as necessary.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Azure IoTEdge PR checklist:
+
+This checklist is used to make sure that common guidelines for a pull request are followed.
+
+### General Guidelines and Best Practices
+- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
+- [ ] Title of the pull request is clear and informative.
+- [ ] Description of the pull request includes a brief summary of the enhancement or bug fix.
```
Comment (damonbarry) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: I love [this](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20). Are "clear and informative" and "brief summary" enough?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (damonbarry) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Also, I like the steps that kodiak takes when we add the 'ready-to-merge' label to a PR:
- PR title becomes commit title
- PR description becomes commit description
- PR is squashed and merged (so the # of commits in the PR, and their individual messages, don't really matter)

I don't think we have to mandate the use of that label, but whether you use it or not the merge commit should look more or less the same.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (scb01) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: Done

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (scb01) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Interesting. Given that kodiak makes the whole PR into a single commit, lets reference that in our PR template.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (scb01) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Yep, I like that link as well :). It is hard to provide that level of guidance in the PR template and not have it become a book. I've added a note at the bottom that describes kodiak's behavior and linked to this article.  Let me know if this addresses what you have in mind.

**Pull request merged**